### ISSUE TITLE
Concept URIs

### ISSUE THREAD

**Issue by bellerophons-pegasus** [acdh-oeaw/vocabseditor#2]
Is there a way to get other concept URIs than <some schema URI>#concept<a number>?

It would be e.g. nice to have an URI based on the label, separated with a slash instead of a hash like e.g. <some schema URI>/label

**Comment by zxenia** [comment]
Hi,
yes, it can be done - to check if there is notation entered in a database and if yes then serialize URIs with appending notation, not a database id as it is now.

**Comment by csae8092** [comment]
added a settings-param to enable the usage of skos:notation in skosconcept URIs... BUT we don't enforce uniqueness of skos:notations (I think), therefore the URI uses now the skos:notation (slugged) and the internal db-ID; also implementing this might break existing vocabs? @zxenia @bellerophons-pegasus  any thoughts?

**Comment by zxenia** [comment]
Yes, we don't enforce skos:notation to be unique and this is desired behaviour - because we might have the exact same notations in two different vocabs. Also skos:notation can be NULL. 
The problem will appear when user will try to assign the same notation to two different concepts in one schema - this can be solved by binding `schema` and `notation` fields to be unique together if notation is not NULL.

**Comment by bellerophons-pegasus** [comment]
Maybe the uniqueness-issue can be tackled as follows: if in the current vocabulary the concept URI is not unique a warning could be displayed to the user and thus allow for either a manual change of the URI or the skos:notation, where the latter does not have to be enforced.
